opencga-analysis/src/main/java/org/opencb/opencga/analysis/sample/qc/SampleQcAnalysis.java

```diff
@@ -131,237 +130,235 @@ protected void check() throws Exception {
-                    && !analysisParams.getVsQuery().toParams().isEmpty()) {
-                new ToolException("Invalid parameters: if variant stats ID is empty, variant stats query must be empty");
+            if (StringUtils.isEmpty(sampleQcParams.getVsId()) && sampleQcParams.getVsQuery() != null
+                    && !sampleQcParams.getVsQuery().toParams().isEmpty()) {
+                throw new ToolException("Invalid parameters: if variant stats ID is empty, variant stats query must be empty");
             }
-            if (StringUtils.isNotEmpty(analysisParams.getVsId())
-                    && (analysisParams.getVsQuery() == null || analysisParams.getVsQuery().toParams().isEmpty())) {
-                new ToolException("Invalid parameters: if you provide a variant stats ID, variant stats query can not be empty");
+            if (StringUtils.isNotEmpty(sampleQcParams.getVsId())
+                    && (sampleQcParams.getVsQuery() == null || sampleQcParams.getVsQuery().toParams().isEmpty())) {
+                throw new ToolException("Invalid parameters: if you provide a variant stats ID, variant stats query can not be empty");
             }
-            if (StringUtils.isEmpty(analysisParams.getVsId())) {
-                analysisParams.setVsId(OPENCGA_ALL);
+            if (StringUtils.isEmpty(sampleQcParams.getVsId())) {
+                sampleQcParams.setVsId(OPENCGA_ALL);
             }
 
-            if (analysisParams.getVsQuery() == null) {
-                new ToolException("Invalid parameters: variant stats query is empty");
+            if (sampleQcParams.getVsQuery() == null) {
+                throw new ToolException("Invalid parameters: variant stats query is empty");
             }
             if (sample.getQualityControl() != null && sample.getQualityControl().getVariant() != null) {
                 if (CollectionUtils.isNotEmpty(sample.getQualityControl().getVariant().getVariantStats())
-                        && OPENCGA_ALL.equals(analysisParams.getVsId())) {
+                        && OPENCGA_ALL.equals(sampleQcParams.getVsId())) {
                     runVariantStats = false;
                 } else {
                     for (SampleQcVariantStats variantStats : sample.getQualityControl().getVariant().getVariantStats()) {
-                        if (variantStats.getId().equals(analysisParams.getVsId())) {
-                            throw new ToolException("Invalid parameters: variant stats ID '" + analysisParams.getVsId()
+                        if (variantStats.getId().equals(sampleQcParams.getVsId())) {
+                            throw new ToolException("Invalid parameters: variant stats ID '" + sampleQcParams.getVsId()
                                     + "' is already used");
                         }
                     }
                 }
             }
+        } else {
+            String msg = "Skipping sample variant stats analysis by user";
+            addWarning(msg);
```

Comment:
Is this even a warning, or should it be just an info event?

Reply:
hard to say!